src/util/simplify_expr.cpp

```diff
@@ -405,6 +474,69 @@ static simplify_exprt::resultt<> simplify_string_char_at(
   return c;
 }
 
+/// Take the passed-in constant string array and lower-case every character.
+static void lower_case_string_expression(array_exprt &string_data)
+{
+  auto &operands = string_data.operands();
+  for(auto &operand : operands)
+  {
+    auto &constant_value = to_constant_expr(operand);
+    char character = numeric_cast_v<int>(constant_value);
```

Comment:
Again, watch out that you're restricting to ASCII here and refuse to const-prop anything outside that range, since we don't have a C++ widget with the same case semantics as Character.toUpperCase